spec/models/manageiq/providers/kubernetes/container_manager/scanning/job_spec.rb

```diff
@@ -144,6 +144,20 @@ def fetch_metadata(*_args)
       end
     end
 
+    it 'should add correct environment variables' do
+      att_name = 'http_proxy'
+      my_value = "MY_TEST_VALUE"
+      @ems.custom_attributes.create(:section => described_class::ATTRIBUTE_SECTION,
+                                    :name    => att_name,
+                                    :value   => my_value)
+      allow_any_instance_of(described_class).to receive_messages(:kubernetes_client => MockKubeClient.new)
```

Comment:
It's terribly complicated rspec code that could break each time we upgrade ruby.  Here is an [example](https://github.com/rspec/rspec-mocks/blob/4d842fefb82eaca7d33554746c1fb8c03738b068/lib/rspec/mocks/any_instance/recorder.rb#L197-L218).  Typically it's because we can't easily instantiate the object we want so we mock any instance of the class.  It's not always easy to remove but it's clearly tech debt you have to be willing to deal with each time we upgrade.